File: Web/src/app/models/search-page-dto.ts
import { environment } from '../../environments/environment';
import { OrderDirection } from './order-direction';

export class SearchPageDto<TDto> {
  constructor() {
    this.pageSize = environment.pageSize;
  };

  criteria: TDto = {} as TDto;
  orderDirection: OrderDirection;
  page: number;
  pageSize: number = environment.pageSize;
  sortBy?: string = environment.sortBy;
}

File: DTO/SearchPageDto.cs
namespace DTO
{
    public class SearchPageDto<T>
    {
        public int Page { get; set; }
        public int PageIndex { get { return Page - 1; } }

        public int PageSize { get; set; }

        public string SortBy { get; set; }

        public OrderDirectionEnum OrderDirection { get; set; }

        public T Criteria { get; set; }
    }
}
